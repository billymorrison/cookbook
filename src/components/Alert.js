import PropTypes from "prop-types";
import styled from "styled-components";

const ErrorBlock = styled.div`
  box-sizing: border-box;
  background-color: lightgreen;
  border: 1px solid green;
  width: 300px;
  height: 40px;
  border-radius: 2px;
  display: flex;
  justify-content: center;
  align-items: center;

  ${({error}) => error && `
      background-color: salmon;
      border: 1px solid red;
    `}
`;

const Alert = ({error, message}) => {
  return (
    <ErrorBlock error={error} data-testid="alert">
      {message}
    </ErrorBlock>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
