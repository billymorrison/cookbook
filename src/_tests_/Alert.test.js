import ReactDOM from "react-dom";
import Alert from "../components/Alert";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

const error = {
    status: 404
}

const message = "Cannot be found"

it("renders alert component without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Alert error={error} message={message}></Alert>, div)
});

it("renders alert correctly", () => {
  const {getByTestId} = render(<Alert error={error} message={message}></Alert>);
  expect(getByTestId("alert")).toHaveTextContent("Cannot be found");
})

it("matches snapshot", () => {
  const tree = renderer.create(<Alert error={error} message={message}></Alert>).toJSON();
  expect(tree).toMatchSnapshot();
})