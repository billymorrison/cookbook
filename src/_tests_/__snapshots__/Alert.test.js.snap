// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdfBwQ jbHHQv"
  data-testid="alert"
>
  Cannot be found
</div>
`;
